packages/graph/hash_graph/tests/integration/postgres/mod.rs

```diff
@@ -0,0 +1,150 @@
+
+    pub fn create_data_type(
+        &mut self,
+        data_type: DataType,
+    ) -> Result<Qualified<DataType>, InsertionError> {
+        self.rt.block_on(async {
+            let data_type = self
+                .postgres
+                .create_data_type(data_type, self.account_id)
+                .await?;
+            self.created_base_uris
+                .push(data_type.inner().id().base_uri().clone());
+            Ok(data_type)
+        })
+    }
+
+    pub fn get_data_type(
+        &mut self,
+        version_id: VersionId,
+    ) -> Result<Qualified<DataType>, QueryError> {
+        self.rt
+            .block_on(async { self.postgres.get_data_type(version_id).await })
+    }
+
+    pub fn update_data_type(
+        &mut self,
+        data_type: DataType,
+    ) -> Result<Qualified<DataType>, UpdateError> {
+        self.rt.block_on(async {
+            self.postgres
+                .update_data_type(data_type, self.account_id)
+                .await
+        })
+    }
+}
+
+impl Drop for Database {
+    fn drop(&mut self) {
+        let mut connection = self.connection.take().unwrap();
+        let account_id = self.account_id;
+        let created_base_uris = self.created_base_uris.clone();
+
+        self.rt.block_on(async move {
+            for base_uri in created_base_uris {
+                remove_by_base_uri(&mut connection, &base_uri).await;
+            }
+            remove_account_id(&mut connection, account_id).await;
+        });
+    }
+}
+
+async fn create_account_id(connection: &mut PgConnection) -> AccountId {
+    let account_id = AccountId::new(Uuid::new_v4());
+    sqlx::query(r#"INSERT INTO accounts (account_id) VALUES ($1);"#)
+        .bind(account_id)
+        .fetch_all(connection)
+        .await
+        .expect("could not insert account id");
+    account_id
+}
+
+async fn remove_account_id(connection: &mut PgConnection, account_id: AccountId) {
+    sqlx::query(r#"DELETE FROM accounts WHERE account_id = $1;"#)
+        .bind(account_id)
+        .execute(connection)
+        .await
+        .expect("could not remove account id");
+}
+
+async fn remove_by_base_uri(connection: &mut PgConnection, base_uri: &BaseUri) {
+    async fn remove_version_id_by_base_uri(connection: &mut PgConnection, base_uri: &BaseUri) {
+        sqlx::query(r#"DELETE FROM version_ids USING ids WHERE ids.version_id = version_ids.version_id AND base_uri = $1;"#)
+            .bind(base_uri)
+            .execute(connection)
+            .await
+            .expect("could not remove version_id for base uri");
+    }
+    async fn remove_base_uri(connection: &mut PgConnection, base_uri: &BaseUri) {
+        sqlx::query(r#"DELETE FROM base_uris WHERE base_uri = $1;"#)
+            .bind(base_uri)
+            .execute(connection)
+            .await
+            .expect("could not remove base_uri");
+    }
+
+    remove_version_id_by_base_uri(connection, base_uri).await;
+    remove_base_uri(connection, base_uri).await;
+}
```

Comment:
`.execute` takes `E: Executor` and `Executor` is implemented for `&mut PgConnection`. This implies that passing &mut PgConnection` "consumes" the reference as it's passed "by value". The inner functions are a workaround to solve this.

Reply:
I don't think I quite understand. What's the error if you just do

```rust
async fn remove_by_base_uri(connection: &mut PgConnection, base_uri: &BaseUri) {
        sqlx::query(r#"DELETE FROM version_ids USING ids WHERE ids.version_id = version_ids.version_id AND base_uri = $1;"#)
            .bind(base_uri)
            .execute(connection)
            .await
            .expect("could not remove version_id for base uri");

        sqlx::query(r#"DELETE FROM base_uris WHERE base_uri = $1;"#)
            .bind(base_uri)
            .execute(connection)
            .await
            .expect("could not remove base_uri");
}
```

Reply:
```
error[E0382]: use of moved value: `connection`
   --> lib/graph/../../tests/integration/postgres/mod.rs:143:18
    |
132 | async fn remove_by_base_uri(connection: &mut PgConnection, base_uri: &BaseUri) {
    |                             ---------- move occurs because `connection` has type `&mut PgConnection`, which does not implement the `Copy` trait
...
136 |             .execute(connection)
    |                      ---------- value moved here
...
143 |         .execute(connection)
    |                  ^^^^^^^^^^ value used here after move
```

Reply:
This works
```rust
async fn remove_by_base_uri(connection: &mut PgConnection, base_uri: &BaseUri) {
    connection.execute(sqlx::query(r#"DELETE FROM version_ids USING ids WHERE ids.version_id = version_ids.version_id AND base_uri = $1;"#)
            .bind(base_uri))
            .await
            .expect("could not remove version_id for base uri");

    connection
        .execute(sqlx::query(r#"DELETE FROM base_uris WHERE base_uri = $1;"#).bind(base_uri))
        .await
        .expect("could not remove base_uri");
}
```

Reply:
I think we should probably stick to the second format generally anyway. I think it's a little weird to call `Query::execute(connection)` compared to `Connection::execute(Query)`

Reply:
I didn't know that was possible as well. I will change this when touching other query-code.